src/commands/reword.rs

```diff
@@ -0,0 +1,358 @@
+    let event_log_db = EventLogDb::new(&conn)?;
+    let event_replayer = EventReplayer::from_event_log_db(effects, &repo, &event_log_db)?;
+    let event_cursor = event_replayer.make_default_cursor();
+    let mut dag = Dag::open_and_sync(
+        effects,
+        &repo,
+        &event_replayer,
+        event_cursor,
+        &references_snapshot,
+    )?;
+
+    let commits = match resolve_commits_from_hashes(&repo, &mut dag, effects, hashes)? {
+        Some(commits) => commits,
+        None => return Ok(1),
+    };
+
+    let messages = match build_messages(&messages, &commits, get_comment_char(&repo)?)? {
+        BuildRewordMessageResult::Succeeded { messages } => messages,
+        BuildRewordMessageResult::IdenticalMessage => {
+            writeln!(
+                effects.get_output_stream(),
+                "The message wasn't edited; nothing to do. Stopping."
+            )?;
+            return Ok(1);
+        }
+        BuildRewordMessageResult::EmptyMessage => {
+            writeln!(
+                effects.get_error_stream(),
+                "Error: the reworded message was empty. Stopping."
+            )?;
+            return Ok(1);
+        }
+    };
+
+    let subtree_roots = find_subtree_roots(&repo, &dag, &commits)?;
+
+    let rebase_plan = {
+        let pool = ThreadPoolBuilder::new().build()?;
+        let repo_pool = RepoResource::new_pool(&repo)?;
+        let mut builder = RebasePlanBuilder::new(&dag);
+
+        for root_commit in subtree_roots {
+            let only_parent_id = root_commit.get_only_parent().map(|parent| parent.get_oid());
+            if only_parent_id == None {
+                eyre::bail!(
+                    "###
+                    Refusing to reword commit {}, which has {} parents.
+                    Rewording only supported for commits with 1 parent.
+                    Stopping; nothing reworded.
+                    ###",
+                    root_commit.get_oid(),
+                    root_commit.get_parents().len(),
+                );
+            }
+            builder.move_subtree(root_commit.get_oid(), only_parent_id.unwrap())?;
+        }
+
+        for commit in commits.iter() {
+            let message = messages.get(&commit.get_oid()).unwrap();
+            // This looks funny, but just means "leave everything but the message as is"
+            let replacement_oid =
+                commit.amend_commit(None, None, None, Some(message.as_str()), None)?;
+            builder.replace_commit(commit.get_oid(), replacement_oid)?;
+        }
+
+        match builder.build(
+            effects,
+            &pool,
+            &repo_pool,
+            &BuildRebasePlanOptions {
+                dump_rebase_constraints: false,
+                dump_rebase_plan: false,
+                detect_duplicate_commits_via_patch_id: false,
+            },
+        )? {
+            Ok(Some(rebase_plan)) => rebase_plan,
+            Ok(None) => {
+                writeln!(
```

Comment:
This is actually a case where we could use `eyre::bail!` 😂 (but this is certainly fine too).